src/picongpu/include/particles/ionization/byField/BSIHydrogenLike/AlgorithmBSIHydrogenLike.hpp

```diff
@@ -64,13 +65,16 @@ namespace ionization
         operator()( const BType bField, const EType eField, ParticleType& parentIon )
         {
 
-            const float_X protonNumber = GetAtomicNumbers<ParticleType>::type::numberOfProtons;
-            float_X chargeState = attribute::getChargeState(parentIon);
-
-            uint32_t cs = math::float2int_rd(chargeState);
+            const float_X protonNumber  = GetAtomicNumbers<ParticleType>::type::numberOfProtons;
+            float_X chargeState         = attribute::getChargeState(parentIon);
+            uint32_t cs                 = math::float2int_rd(chargeState);
+            /* ionization potential in atomic units */
+            const float_X iEnergy       = GetIonizationEnergies<ParticleType>::type()[cs];
```

Comment:
If the ion is fully ionized and `chargeState` is `protonNumber` this results in an invalid memory access. 
Ionization energies are only defined for the index `[0,protonNumber-1]`